cubi_tk/snappy/itransfer_common.py

```diff
@@ -329,7 +332,190 @@ def build_jobs(self, library_names) -> typing.Tuple[TransferJob, ...]:
                             bytes=size,
                         )
                     )
-        return tuple(sorted(transfer_jobs))
+        return lz_uuid, tuple(sorted(transfer_jobs))
+
+    def get_sodar_info(self):
+        """
+        Method evaluates user input to extract or create iRODS path. Use cases:
+        1. User provides iRODS path. Same as before, use it.
+        2. User provides Landing Zone UUID. Same as before, fetch path and use it.
+        3. User provides Project UUID:
+           i. If there are LZ associated with project, select the latest active and use it.
+          ii. If there are no LZ associated with project, create a new one and use it.
+        4. Data provided by user is neither an iRODS path nor a valid UUID. Report error and exit.
+
+        :return: Returns landing zone UUID and path to iRODS directory.
+        """
+        # Initialise variables
+        lz_irods_path = None
+        lz_uuid = None
+        is_project_uuid = True
+        is_lz_uuid = True
+        create_lz_bool = self.args.yes
+        in_destination = self.args.destination
+
+        # iRODS path provided by user
+        # Not possible to retrieve lz uuid from path. Returns None for lz_uuid.
+        if "/" in in_destination:
+            lz_irods_path = in_destination
+
+        # Project UUID provided by user
+        elif is_uuid(in_destination):
+
+            if create_lz_bool:
+                # Assume that provided UUID is associated with a Project and user wants a new LZ.
+                # Behavior: create new LZ.
+                try:
+                    lz_uuid, lz_irods_path = self.create_landing_zone(project_uuid=in_destination)
+                except requests.exceptions.HTTPError as e:
+                    logger.error("Unable to create Landing Zone using UUID {uuid}. "
+                                 "HTTP error {error}".format(uuid=in_destination, error=str(e)))
+                    exit(os.EX_DATAERR)
```

Comment:
The pythonic way is to use exceptions here. Look at the existing `cubi_tk.exceptions` module.

Reply:
Changed.